src/query/execute_statement.c

```diff
@@ -14532,118 +14491,155 @@ do_prepare_session_statement (PARSER_CONTEXT * parser, PT_NODE * statement)
  * do_prepare_cte () - prepare CTE statements in WITH clause for query cache
  * return : Error code
  * parser (in)	  : parser context
- * statement (in) : statement to prepare
+ * stmt (in) : statement to prepare
  */
-static int
-do_prepare_cte (PARSER_CONTEXT * parser, PT_NODE * statement)
+int
+do_prepare_cte (PARSER_CONTEXT * parser, PT_NODE * stmt)
 {
-  int error = NO_ERROR;
-  PT_NODE *cte_statement;
-  PT_NODE *cte_def_list;
+  int err = NO_ERROR;
   PARSER_CONTEXT cte_context;
-  int var_count;
+  int i, var_count;
+
+  cte_context = *parser;
```

Comment:
Rather than referencing an existing `parser`, a new `PARSER_CONTEXT `object is being created.
Shouldn't we clean up the memory allocated to this object?

Reply:
It's just initialization by existing parser context.

Reply:
Since most of the context in the parser is not used and COMPILE_CONTEXT in the parser is newly allocated and used, there does not seem to be a problem.
And since the CTE query is first prepared and cached in xasl before the main query is processed, there seems to be no problem with the main query processing thereafter.